gluten-data/src/main/java/io/glutenproject/columnarbatch/ColumnarBatches.java

```diff
@@ -130,6 +134,39 @@ public static ColumnarBatch ensureLoaded(BufferAllocator allocator, ColumnarBatc
     return load(allocator, batch);
   }
 
+  /** The input vsr will be closed after applying this method */
+  public static ColumnarBatch offloadedAsVeloxColumnarBatch(
+      BufferAllocator allocator, VectorSchemaRoot vsr, long memoryManagerHandle) {
+    try (ArrowArray cArray = ArrowArray.allocateNew(allocator);
+        ArrowSchema cSchema = ArrowSchema.allocateNew(allocator)) {
+      ArrowAbiUtil.exportFromVectorSchemaRoot(
+          ArrowBufferAllocators.contextInstance(), vsr, cSchema, cArray);
+      long handle =
+          VeloxColumnarBatchBuilderJniWrapper.INSTANCE.createVeloxColumnarBatch(
+              memoryManagerHandle, cSchema.memoryAddress(), cArray.memoryAddress());
+      return ColumnarBatches.create(handle);
+    }
+  }
+
+  /** Convert input `ColumnarBatch` to `VectorSchemaRoot` and close the input `ColumnarBatch` */
+  public static VectorSchemaRoot loadArrowVectorSchemaRoot(
+      BufferAllocator allocator, ColumnarBatch input, Schema arrowSchema) {
+    ensureOffloaded(allocator, input);
+    IndicatorVector iv = (IndicatorVector) input.column(0);
+    final long handle = iv.getNativeHandle();
+    try (ArrowSchema cSchema = ArrowSchema.allocateNew(allocator);
+        ArrowArray cArray = ArrowArray.allocateNew(allocator)) {
+      ColumnarBatchJniWrapper.INSTANCE.exportToArrow(
+          handle, cSchema.memoryAddress(), cArray.memoryAddress());
+      VectorSchemaRoot vsr = VectorSchemaRoot.create(arrowSchema, allocator);
+      ArrowAbiUtil.importToVectorSchemaRoot(allocator, vsr, cArray);
+      for (long i = 0; i < iv.refCnt(); i++) {
+        input.close();
+      }
+      return vsr;
+    }
+  }
+
```

Comment:
I assume the purpose of adding the new APIs is to implement a zero-copy way to convert Arrow data from/to Array[Byte], is that correct?

Reply:
yeah, I read the related code about Spark Arrow and Gluten. I think it's the shortest path to do the data transform...

Reply:
OK but it's probably not possible to do this without copy. So copy is here and it's OK to me.

And I realized we once added `ColumnarBatchSerializerJniWrapper` as a Java-side serde for columnar batch. Would you think that will be helpful here, since we already can't avoid copies?